46.md

```diff
@@ -2,161 +2,67 @@ NIP-46
-* **Signer**: Nostr app that holds the private key of a nostr account and *can sign* on its behalf.
-
-
-## `TL;DR`
-
-
-**App** and **Signer** sends ephemeral encrypted messages to each other using kind `24133`, using a relay of choice. 
-
-App prompts the Signer to do things such as fetching the public key or signing events.
+```
+<npub1...>#<secret>?relay=wss://...&relay=wss://...
+```
 
-The `content` field must be an encrypted JSONRPC-ish **request** or **response**.
+The user copies that token and pastes it in the client UI somehow. Then the client can send events of kind `24133` to the specified relays and wait for responses from the remote signer.
 
-## Signer Protocol
+The `secret` allows the signer to identify the client's key and, perhaps, assign it certain permissions (e.g. *sign only event `kind:7`, reject encrypt/decrypt requests*)
 
-### Messages
+Event payloads are [NIP-04](04.md)-encrypted JSON blobs that look like JSONRPC.
 
-#### Request
+Events sent by the client to the remote signer have the following format:
 
-```json
+```js
 {
-  "id": <random_string>,
-  "method": <one_of_the_methods>,
-  "params": [<anything>, <else>]
+  "pubkey": "<client-deterministically-generated-key-for-the-connection>"
+  "kind": 24133,
+  "tags": [
+    ["p", "<user-pubkey-hex>"]
+  ],
+  "content": "nip04_encrypted_json({id: <random-string>, method: <see-below>, params: []})",
+  ...
 }
 ```
 
-#### Response
+And the events the remote signer sends to the client have the following format:
 
-```json
-{
-  "id": <request_id>,
-  "result": <anything>,
-  "error": <reason>
-}
+```js
+  "pubkey": "<user-pubkey>"
+  "kind": 24133,
+  "tags": [
+    ["p", "<signer-pubkey-hex>"]
+  ],
+  "content": "nip04_encrypted_json({id: <request-id>, result: <any>, error: <reason-string>})",
+  ...
 ```
 
 ### Methods
 
-
-#### Mandatory 
-
-These are mandatory methods the remote signer app MUST implement:
-
-- **describe**
-  - params []
-  - result `["describe", "get_public_key", "sign_event", "connect", "disconnect", "delegate", ...]`  
+- **connect**
+  - params: [`pubkey`, `secret`]
+  - result: `null`
 - **get_public_key**
-  - params []
-  - result `pubkey` 
+  - params: []
+  - result: `pubkey`
 - **sign_event**
-  - params [`event`]
-  - result `event_with_signature` 
-
-#### optional
-
-
-- **connect**
-  - params [`pubkey`]
-- **disconnect**
-  - params []
-- **delegate** 
-  - params [`delegatee`, `{ kind: number, since: number, until: number }`]
-  - result `{ from: string, to: string, cond: string, sig: string }`
+  - params: [`event`]
+  - result: `event_with_pubkey_id_and_signature`
 - **get_relays**
-  - params []
-  - result `{ [url: string]: {read: boolean, write: boolean} }` 
+  - params: []
+  - result: `{ [url: string]: {read: boolean, write: boolean} }`
 - **nip04_encrypt**
-  - params [`pubkey`, `plaintext`]
-  - result `nip4 ciphertext`
+  - params: [`pubkey`, `plaintext`]
```

Comment:
for the nip04_encrypt and nip04_decrypt methods below, we should update params to contain `peer pubkey` or some other description that makes it clear it is not the user's own pubkey

Reply:
I am so confused by this I won't update it until I actually know what I'm doing, but please send a PR if you do.